client/components/sites-add-new-site/test/index.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * @jest-environment jsdom
+ */
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import React from 'react';
+import { SitesAddNewSitePopover } from '../index';
+
+// Mock dependencies
+jest.mock( '@automattic/calypso-analytics', () => ( {
+	recordTracksEvent: jest.fn(),
+} ) );
+
+jest.mock( 'i18n-calypso', () => ( {
+	useTranslate: () => ( text: string ) => text,
+	localize: ( x ) => x,
+} ) );
+
+// Mock the AsyncContent component to avoid loading issues in tests
+jest.mock( '../content/async', () => ( {
+	AsyncContent: () => <div data-testid="async-content">Async Content</div>,
+} ) );
+
+describe( 'SitesAddNewSitePopover', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'renders with full label when not in compact mode', () => {
+		render( <SitesAddNewSitePopover showCompact={ false } /> );
+		expect( screen.getByText( 'Add new site' ) ).toBeInTheDocument();
+	} );
+
+	it( 'renders without label in compact mode', () => {
+		render( <SitesAddNewSitePopover showCompact /> );
+		expect( screen.queryByText( 'Add new site' ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'opens popover and tracks event when button is clicked', () => {
+		render( <SitesAddNewSitePopover showCompact={ false } /> );
+		const button = screen.getByRole( 'button' );
+		fireEvent.click( button );
+
+		expect( screen.getByTestId( 'async-content' ) ).toBeInTheDocument();
+		expect( recordTracksEvent ).toHaveBeenCalledWith(
+			'calypso_sites_dashboard_new_site_action_click_open'
+		);
+	} );
+
+	it( 'closes popover when clicking outside', async () => {
+		render( <SitesAddNewSitePopover showCompact={ false } /> );
+		fireEvent.click( screen.getByRole( 'button' ) );
+		expect( screen.getByTestId( 'async-content' ) ).toBeInTheDocument();
+
+		fireEvent.click( document.body );
+		waitFor( () => {
```

Comment:
I tripped over this yesterday - I think we need to `await` the result of `waitFor`.
```suggestion
		await waitFor( () => {
```

Reply:
Ahk! Thanks.